css-cascade-6/Overview.bs

```diff
@@ -51,6 +51,227 @@ Introduction and Missing Sections</h2>
+-->
+
+<h2 id="at-import">
+Importing Style Sheets: the ''@import'' rule</h2>
+
+	The <dfn>@import</dfn> rule allows users to import style rules from other style sheets.
+	If an ''@import'' rule refers to a valid stylesheet,
+	user agents must treat the contents of the stylesheet as if they were written in place of the ''@import'' rule,
+	with two exceptions:
+
+	* If a feature
+		(such as the ''@namespace'' rule)
+		<em>explicitly</em> defines that it only applies to a particular stylesheet,
+		and not any imported ones,
+		then it doesn't apply to the imported stylesheet.
+
+	* If a feature relies on the relative ordering of two or more constructs in a stylesheet
+		(such as the requirement that ''@namespace'' rules must not have any other rules other than
+		''@import'' preceding it),
+		it only applies between constructs in the same stylesheet.
+
+	<p class='example'>
+		For example, [=declarations=] in style rules from imported stylesheets interact with the cascade
+		as if they were written literally into the stylesheet at the point of the ''@import''.
+
+	Any ''@import'' rules must precede all other valid at-rules and style rules in a style sheet
+	(ignoring ''@charset'' and <a href="#layer-empty"><css>@layer</css> statement</a> rules)
+	and must not have any other valid at-rules or style rules between it and previous ''@import'' rules,
+	or else the ''@import'' rule is invalid.
+	The syntax of ''@import'' is:
+
+<pre class='prod'>
+		@import [ <<url>> | <<string>> ]
+		        [[ layer | layer(<<layer-name>>) ]
+		         || [ scope( [ (<<scope-start>>) ]? [ to (<<scope-end>>) ]? | <<scope-start>> ) ]
+		         || <<supports-import-condition>>]?
+		        <<media-import-condition>>?;
+
+		<dfn export>&lt;supports-import-condition></dfn> = [ supports( [ <<supports-condition>> | <<declaration>> ] ) ]
+		<dfn export>&lt;media-import-condition></dfn> = <<media-query-list>></pre>
+
+	where:
+
+	* the <<url>> or <<string>>
+		gives the URL of the style sheet to be imported.
+
+	and optionally:
+
+	* the ''layer'' keyword or ''layer()'' function,
+		which assigns the contents of the style sheet
+		into its own anonymous [=cascade layer=]
+		or into the named [=cascade layer=].
+
+		The layer is added to the [[#layer-ordering|layer order]]
+		even if the import fails to load the stylesheet,
+		but is subject to any [=import conditions=]
+		(just as if declared by an ''@layer'' rule wrapped
+		in the appropriate [=conditional group rules=]).
+	* the ''scope()'' function,
+		which [=scopes=] the [=style rules=] within the stylesheet,
+		using the [=scoping roots=] and [=scoping limits=]
+		as described by [[#scope-limits]].
+
+		Note: While the [=style rules=] within the imported stylesheet
+		become [=scoped=],
+		they do not become [=nested style rule|nested=].
+		In particular,
+		top-level selectors are not re-interpreted as [=relative selectors=],
+		and the ''&'' pseudo-class maintains its non-nested behavior.
```

Comment:
I see the reasoning behind this, and think it's probably the best behavior - but it is a departure from `@scope`. I image we would also apply this behavior to any html `scope` attribute, so those would be consistent. Is it something that deserves it's own discussion?

Reply:
Is it correct that wrapping an entire stylesheet's contents with `@scope` would behave differently from importing that same stylesheet with a `scope()` condition?

I am unsure if that is the best option.
I think it would be surprising to authors as they would assume this to be purely a syntactic difference.

If the behavior is different it also implies that bundling becomes impossible, which might make the entire feature unusable in practice as it would force CSS authors to serve unbundled stylesheets and take the associated performance hit.

Reply:
Fair points. It sounds like this does in fact deserve it's own issue. :)

Reply:
> I see the reasoning behind this

I'm not sure I do (anymore). I remember concluding that anything else was impossible, but I don't remember why.

We'd have to allow relative selectors top-level, but we could do that and spec that they are relative against `:scope`.

(Re-interpreting `&` is deeply incompatible with Blink's implementation of nesting, but we don't have to let that guide the discussion.)

Reply:
Resolved in https://github.com/w3c/csswg-drafts/issues/11756.